### PR TITLE
fix(kube-nodes): flavor update post node creation

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
Close MBP-308

### Requirements

After a node is added, the flavor is not updated correctly.

### Description of the Change

The issue is fixed by updating the flavor by getting the list of flavors and filtering them by the flavor code from the node.

### CODE REVIEW

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 1]
```diff
@@ -56,15 +56,16 @@ angular.module('managerApp').controller('KubernetesNodesCtrl', class KubernetesN
   }
 
   getAssociatedFlavor(node) {
-    if (node.instanceId) {
-      return this.Kubernetes.getAssociatedInstance(node.projectId, node.instanceId)
```
Comment (antleblanc) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: Is `getAssociatedInstance` method from the `Kubernetes` still need? 

```sh
$ git grep -i 'getAssociatedInstance'
client/app/kubernetes/kubernetes.service.js:  getAssociatedInstance(projectId, instanceId) {
client/app/kubernetes/nodes/kubernetes-nodes.controller.js:      return this.Kubernetes.getAssociatedInstance(node.projectId, node.instanceId)
```

Comment (ganeshkumar1989) [reply client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: This method is no longer being used anywhere. Hence it has been removed, Thanks for pointing it out :)

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 2]
```diff
@@ -56,15 +56,16 @@ angular.module('managerApp').controller('KubernetesNodesCtrl', class KubernetesN
...
-        .then(instance => _.set(node, 'formattedFlavor', this.Kubernetes.formatFlavor(instance.flavor)))
-        .catch(() => {
-          _.set(node, 'formattedFlavor', this.$translate.instant('kube_nodes_flavor_error'));
-        });
-    }
-
-    return this.$q.when(_.set(node, 'formattedFlavor', node.flavor));
+    return this.Kubernetes.getFlavors(node.projectId)
+      .then((flavors) => {
+        const matchedFlavor = _.find(flavors, flavor => flavor.name === node.flavor);
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]:
```suggestion
        const matchedFlavor = _.find(flavors,  { name: node.flavor });
```

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 3]
```diff
@@ -56,15 +56,16 @@ angular.module('managerApp').controller('KubernetesNodesCtrl', class KubernetesN
...
+        _.set(node, 'formattedFlavor', matchedFlavor
+          ? this.Kubernetes.formatFlavor(matchedFlavor)
+          : this.$translate.instant('kube_nodes_flavor_error'));
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: Please use two different messages

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js [hunk 4]
```diff
@@ -56,15 +56,16 @@ angular.module('managerApp').controller('KubernetesNodesCtrl', class KubernetesN
...
+      })
+      .catch(() => {
+        _.set(node, 'formattedFlavor', this.$translate.instant('kube_nodes_flavor_error'));
```
Comment (FredericEspiau) [inline client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: Please use two different messages

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js (hunk 2 above)
Comment (ganeshkumar1989) [reply client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: This is done. Thanks!

File: client/app/kubernetes/nodes/kubernetes-nodes.controller.js (hunk 3 above)
Comment (ganeshkumar1989) [reply client/app/kubernetes/nodes/kubernetes-nodes.controller.js]: There would definitely be a matching flavor and hence the error message is removed here.

**Pull request merged**